oap-server/analyzer/event-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/event/EventAnalyzerServiceImpl.java

```diff
@@ -34,8 +36,20 @@ public class EventAnalyzerServiceImpl implements EventAnalyzerService, EventAnal
 
     @Override
     public void analyze(final Event event) {
+        final Event.Builder eb = event.toBuilder();
+        if (event.getStartTime() <= 0) {
+            log.warn("Event start time {} is invalid, it will be set to current time, eventId: {}",
+                event.getStartTime(), event.getUuid());
+            eb.setStartTime(System.currentTimeMillis());
+        }
+        if (event.getEndTime() <= 0) {
+            log.warn("Event end time {} is invalid, it will be set to current time, eventId: {}",
+                event.getEndTime(), event.getUuid());
+            eb.setEndTime(System.currentTimeMillis());
+        }
+
```

Comment:
I remember you mentioned, start time and endpoint could be assigned separatedly in two requests?

This warning maybe trigger false warning?

Reply:
The think the rule should be either start or end is legal


And these two variables here are timestamps, not time bucket, right?

Reply:
> The think the rule should be either start or end is legal

Ha, right.

> And these two variables here are timestamps, not time bucket, right?

Yes. They are timestamp here, haven't been converted to time bucket yet,